flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java

```diff
@@ -264,15 +278,21 @@ protected void initChannel(SocketChannel socketChannel) {
                     }
                 };
 
-        // No NioEventLoopGroup constructor available that allows passing nThreads, threadFactory,
-        // and selectStrategyFactory without also passing a SelectorProvider, so mimicking its
-        // default value seen in other constructors
-        NioEventLoopGroup group =
-                new NioEventLoopGroup(
-                        1,
-                        new ExecutorThreadFactory("flink-rest-client-netty"),
-                        SelectorProvider.provider(),
-                        selectStrategyFactory);
+        if (group == null) {
+            // No NioEventLoopGroup constructor available that allows passing nThreads,
+            // threadFactory,
+            // and selectStrategyFactory without also passing a SelectorProvider, so mimicking its
+            // default value seen in other constructors
+            group =
+                    new NioEventLoopGroup(
+                            1,
+                            new ExecutorThreadFactory("flink-rest-client-netty"),
+                            SelectorProvider.provider(),
+                            selectStrategyFactory);
+            useInternalEventLoopGroup = true;
+        } else {
+            useInternalEventLoopGroup = false;
```

Comment:
should we check that the supplied group is not shutting down or shutdown

Reply:
I see the channel is `NioSocketChannel.class` is this always the case for external groups?

Reply:
I think the extenal service doesn't care the type of SocketChannel.  As the jira says, it just pass a shared event group to avoid heap memory leak. 

Maybe it's a better choice to specify the contruct param `group` as NioEventLoopGroup, which could  avoid the type error building `Bootstrap` instance ?

Reply:
> should we check that the supplied group is not shutting down or shutdown

This check is necessary and the logic has been added. Thanks